esm_analysis/carbon.py

```diff
@@ -256,27 +256,85 @@ def _check_variables(ds):
     return sensitivity
 
 
+@check_xarray(0)
+def co2_flx_ocean_sensitivities(ds):
+    """Compute sensitivity of oceanic co2_flux to changes in driver variables.
+
+    .. math: co2_flx_ocean = co2trans * (co2atm - co2ocean) * (1 - ice)
+
+    Args:
+        ds (xr.Dataset): containing cmorized variables:
+                         * co2_flx_ocean [kg m-2 s-1]: surface air-sea CO2 flux
+                         * co2ocean[ppm CO2]: partial pressure of CO2 in
+                                              surface sea water
+                         * CO2[ppm]: partial pressure of atmospheric CO2 at the
+                                     surface
+                         * co2trans [10-9 mol s m-4] : transfer velocity of
+                                                       ocean/atmosphere CO2 flux
+                         * seaice [%]: fraction of grid cell covered by ice
+
+    Returns:
+        sensitivity (xr.Dataset):
+
+    References:
+        * Lovenduski, Nicole S., Nicolas Gruber, Scott C. Doney, and Ivan D. Lima.
+          “Enhanced CO2 Outgassing in the Southern Ocean from a Positive Phase of
+          the Southern Annular Mode.” Global Biogeochemical Cycles 21, no. 2
+          (2007). https://doi.org/10/fpv2wt.
+        * Sarmiento, Jorge Louis, and Nicolas Gruber. Ocean Biogeochemical Dynamics.
+          Princeton, NJ: Princeton Univ. Press, 2006., p.421, eq. (10:3:1)
+
+    """
+
+    def _check_variables(ds):
+        requiredVars = ['co2_flx_ocean', 'CO2', 'co2ocean', 'co2trans', 'seaice']
+        if not all(i in ds.data_vars for i in requiredVars):
+            missingVars = [i for i in requiredVars if i not in ds.data_vars]
+            raise ValueError(
+                f"""Missing variables needed for calculation:
+            {missingVars}"""
+            )
+
+    _check_variables(ds)
+
+    # Sensitivities are based on the time-mean for each field. This computes
+    # sensitivities at each grid cell.
+    # TODO: Add keyword for sliding mean, as in N year chunks of time to
+    # account for trends.
+    CO2FLUX = ds['co2_flx_ocean']
+    CO2OCEAN = ds['co2ocean']
+    CO2ATM = ds['CO2']
+    ICE = ds['seaice']
+    CO2TRANS = ds['co2trans']
+
+    # Compute sensitivities
```

Comment:
We need specific sensitivities here, right? As in pCO2, where we say:

```

    # Compute sensitivities
    sensitivity = dict()
    sensitivity['tos'] = 0.0423
    sensitivity['sos'] = 1 / SALT
    sensitivity['talkos'] = (1 / ALK) * buffer_factor['ALK']
    sensitivity['dissicos'] = (1 / DIC) * buffer_factor['DIC']
    sensitivity = xr.Dataset(sensitivity) * pCO2
    return sensitivity
```

Reply:
The problem is that this is model dependent. You need to know the CO2 flux equation going in, whether it uses U2, U3, e.g. parameterization for gas flux.

CESM uses U2 for instance.

![Screen Shot 2019-09-02 at 5 38 17 PM](https://user-images.githubusercontent.com/8881170/64135707-28886680-cda0-11e9-993c-2dcc2a587a61.png)

Really what one needs is dF/dpco2 since you then multiply all of that by the pCO2 sensitivities we already have here.

![Screen Shot 2019-09-02 at 5 40 34 PM](https://user-images.githubusercontent.com/8881170/64135728-64bbc700-cda0-11e9-85d7-506f2f66e728.png)

Reply:
Anyways perhaps it's best to have a keyword for the model being used. We could store a dictionary that says CESM is U2, MPI is ____, and so on. That would effect the dF/dpCO2 that's used.

This isn't an easy problem!

Reply:
MPI-ESM also uses a u**2 version. I like the idea of having model dependent sensitivities and therefore have different dictionaries for our models. why dont we have `co2_flx_ocean_sensitivities` for the MPI-ESM and `fgco2_sensitivities` for CESM? for these sensitivties a documentation page would be useful anyways at some point.

Reply:
This sounds like a nice solution. Since this is a package for us and colleagues, when we use a new model we can add it to the dictionary. (E.g. if I need to do this for my MPAS output).

Reply:
A documentation page on this would be good for us too. What do you envision? Perhaps the CO2 flux equation for each model we're working with. As well as variable names. And the sensitivities based on the equation, like in my above screenshots.